hardware_interface/include/hardware_interface/types/hardware_interface_return_values.hpp

```diff
@@ -26,6 +28,22 @@ enum class return_type : std::uint8_t
   DEACTIVATE = 2,
 };
 
+/**
+ * Struct to store the status of the Hardware read or write methods return state.
+ * The status contains information if the cycle was triggered successfully, the result of the
+ * cycle method and the execution duration of the method. The status is used to provide
+ * feedback to the controller_manager.
+ * @var successful: true if it was triggered successfully, false if not.
+ * @var result: return_type::OK if update is successfully, otherwise return_type::ERROR.
+ * @var execution_time: duration of the execution of the update method.
+ */
+struct HardwareComponentCycleStatus
+{
+  bool successful = true;
+  return_type result = return_type::OK;
+  std::optional<std::chrono::nanoseconds> execution_time = std::nullopt;
```

Comment:
the fact that I had to look at the documentation for what `successful` and `result` mean means we should improve the naming here. Let's do that in a follow-up PR though

Reply:
Sure 🫡

Reply:
This was a part of the statistics that's why it isn't clear. I can make the changes in follow-up PR 
Thanks